modules/frontend/frontend.go

```diff
@@ -265,12 +253,26 @@ func buildUpstreamRequestURI(originalURI string, params url.Values) string {
 	return uri
 }
 
-func newQueryRangeMiddleware(cfg Config, o overrides.Interface, reader tempodb.Reader, logger log.Logger) Middleware {
-	return MiddlewareFunc(func(next http.RoundTripper) http.RoundTripper {
-		searchRT := NewRoundTripper(next, newQueryRangeSharder(reader, o, cfg.Metrics.Sharder, newSearchProgress, logger))
+func newQueryRangeMiddleware(cfg Config, o overrides.Interface, reader tempodb.Reader, logger log.Logger) pipeline.Middleware {
+	return pipeline.MiddlewareFunc(func(next http.RoundTripper) http.RoundTripper {
+		searchRT := NewRoundTripper(next, newQueryRangeSharder(reader, o, cfg.Metrics.Sharder, logger))
 
-		return RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		return pipeline.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
 			return searchRT.RoundTrip(r)
 		})
 	})
 }
+
+func asyncPipeline(cfg Config, sharder pipeline.AsyncMiddleware[*http.Response], logger log.Logger) []pipeline.AsyncMiddleware[*http.Response] {
+	if cfg.MultiTenantQueriesEnabled {
```

Comment:
I think this might be changing multi-tenant behavior unintentionally.  Previously when `cfg.MultiTenantQueriesEnabled==false` it would pass-through to next().  But now it looks like it will fail whenever multiple tenant IDs are given.   

For example if `cfg.MultiTenantQueriesEnabled==false` and `X-Scope-OrgID: foo|bar` it will fail instead of using tenantID `foo|bar` literally?

Reply:
yup, you are correct. fixed to return a "no-op" middleware when `MultiTenantQueriesEnabled==false`. this will cause the pipeline to treat `foo|bar` literally.